### PR TITLE
MONGOID-5312 Document attributes method in reference

### PR DESCRIPTION
**Pull request by Neilshweky**

### CODE REVIEW

File: docs/reference/crud.txt [hunk 1]
```diff
@@ -37,6 +37,31 @@ operations with examples.
    * - Operation
      - Example
 
+   * - ``Model#attributes``
+
+       *Returns a hash of the document's attributes with its values in
+       mongoized form (i.e. the way it's stored in the db).*
```
Comment (p-mongo) [inline docs/reference/crud.txt:43]:
```suggestion
       mongoized form (i.e. the way they are stored in the db).*
```

File: docs/reference/crud.txt [hunk 2]
```diff
@@ -37,6 +37,31 @@ operations with examples.
...
+
+       *In Mongoid 8.0 by default, and Mongoid 7.5 with the*
+       ``legacy_attributes`` *flag set to false, the object returned from this
+       method will have class* ``Hash``, *and have string keys. In Mongoid 7.5
+       by default, and in Mongoid 8.0 with the* ``legacy_attributes`` *flag set
+       to true, the object returned from this method will be a*
+       ``BSON::Document`` *(and thus, indifferently accessable) if the document
```
Comment (p-mongo) [inline docs/reference/crud.txt:50]:
```suggestion
       ``BSON::Document`` *(and thus, indifferently accessible) if the document
```

File: docs/reference/crud.txt [hunk 3]
```diff
@@ -37,6 +37,31 @@ operations with examples.
...
+       has been retrieved from the database, and a* ``Hash`` *with string keys
+       if it is a new document.*
+
+       *If an embedded association is empty, its key will not show up in the
+       returned hash.*
```
Comment (p-mongo) [inline docs/reference/crud.txt:55]: Could you add before this sentence, in this paragraph, something like

> The `attributes` hash also contains the attributes of all embedded documents.

If I call `attributes` on a document which is itself embedded and it has another document embedded in it do I get the 2nd and 3rd level (counting from ultimate parent) attributes back?

**Comment by Neilshweky** [comment]
Is that better @p-mongo ?

**Review by p-mongo — APPROVED** [review]

**Review by comandeo — APPROVED** [review]

**Pull request merged**